__tests__/parsers/json-parser.test.ts

```diff
@@ -11,16 +11,20 @@ const content = JSON.stringify(
                 msg2: "world"
             },
         },
-        msg3: "Doe"
+        msg3: "Doe",
+        "msg.4": "this is a fullstop.",
+        "msg.5.": "this is a fullstop. With another sentance."
     }, null, '\t');
 
 test('JSON PARSER: correctly parses from string', async () => {
     const file = await parser.parseFrom(content);
     expect(file).toBeTruthy();
-    expect(file['messages.foo.msg1']).toEqual('hello {{variable}}');
-    expect(file['messages.foo.msg2']).toEqual('world');
-    expect(file['messages.bar']).toEqual('John');
+    expect(file['messages[--]foo[--]msg1']).toEqual('hello {{variable}}');
```

Comment:
It might be nice to have a const `DELIMITER` defined on the `JsonParser` type, so that it can be used consistently.

Reply:
Let me add that...